### PR TITLE
Updates plan conflict notice to accommodate more than one conflict

### PR DESCRIPTION
**Pull request by ChaosExAnima**
#### Changes proposed in this Pull Request

* Updates `< ProductPlanOverlapNotices />` block to accommodate more than one conflict.
* Changes references to just backups to use all Jetpack products.

#### Testing instructions

* Create a Jetpack site with a plan with two or more conflicting products. For example, Jetpack Professional and Scan and Daily Backups.
* Check notices in plans (`/plans/[site]`).
* Check notices in purchases (`/me/purchases`).
* Check notices in site purchases (`/me/purchases/[site]/[purchase]`).

Screen Shot 2020-03-27 at 2 20 17 PM
Screen Shot 2020-03-27 at 2 27 39 PM
Screen Shot 2020-03-27 at 2 28 37 PM
Screen Shot 2020-03-27 at 3 00 07 PM

Fixes 1151678672052943-as-1168265713258996.

### CODE REVIEW

**Comment by ChaosExAnima** [comment]
@keoshi - Thoughts on how I handled multiple conflicts? I thought about having it all in one notice, but that is pretty complex from an i18n perspective and I don't think it'd be a super common case.

**Comment by keoshi** [comment]
@ChaosExAnima I don't mind seeing two blocks. I think this scenario somewhat warrants it (even though combining them would look a lot cleaner — very quick mock:

image

I do think we could tweak the language a little bit to make this clearer. For some reason “Your Jetpack Premium Plan includes scan” is not making a lot of sense to me since the feature appears a bit of out context. What if we replaced it for the main benefit like we're already doing with “daily backups”?`Your Jetpack Premium Plan includes automatic scanning.`

> Looks like you also purchased the %(productName)s product.

This could also probably be:

> Looks like you also purchased the %(productName)s product separately.

Does that make sense? Feel free to add the `Needs Copy Review` for Editorial to chime in if you feel it needs more tweaking.

One last thing: could we replace the `is-warning` status for something less severe? We should be informing customers about this situation so they can take action, but personally "no status" or even the `is-success` with the `gridicons-plans` would convey a much more positive message.

Hope this helps and hope I didn't add a bunch more stuff to your plate.

**Comment by ChaosExAnima** [comment]
@keoshi - That is a good fix! Let's get a 👍 from editorial just in case, though.

**Comment by enejb** [comment]
I think this works really well expect for one case. 
And the case is when we are showing the notice(s) on the individual product manage purchase page. 

This is what I see. 
Screen Shot 2020-03-30 at 12 36 05 PM

I think we should only show the relevant notices here. 
That to me is Just showing the one that is related to scan and the professional plan.

**Comment by michelleweber** [comment]
Good calls on the copy, @keoshi -- let's go with:

> Your Jetpack Premium Plan includes automatic scanning. It also looks like you're using Jetpack Scan as a stand-alone product -- you might want to remove that now.

On the "consider removing" / "you might want to remove" -- do we want to be more directive than that? Like "We recommend removing" or "you should remove"?

**Comment by ChaosExAnima** [comment]
@keoshi @michelleweber - I want to avoid language like `Your Jetpack Premium Plan includes X`, as I'm worried about how to handle i18n for sentences with that sort of structure, especially if there's three or more items in conflict. I'm going to update things to follow the list example @keoshi mocked up.

Regarding action, I **do** agree we're far too passive there. How do users normally resolve this? Can we add a "click here to resolve" action?

@enejb - With the list form, I think it'd make sense to list all even when on a single purchase screen to be consistent. Will post a screen once I have it done.

**Comment by ChaosExAnima** [comment]
Okay! I've resolved all issues. Here's the new design. Note that the links are to manage the product purchases, so the user can remove them. Let me know your thoughts!

Screen Shot 2020-03-31 at 3 13 45 PM

File: client/blocks/product-plan-overlap-notices/index.jsx [hunk 1]
```diff
@@ -99,52 +92,52 @@ class ProductPlanOverlapNotices extends Component {
...
+								'Consider removing conflicting products.',
 							{
 								args: {
-									featureName: this.getOverlappingFeatureName(),
 									planName: this.getCurrentPlanName(),
-									productName: this.getCurrentProductName(),
+								},
+								components: {
+									list: (
+										<ul>
```
Comment (jsnmoon) [inline client/blocks/product-plan-overlap-notices/index.jsx:135]: Suggestion: We might want to apply some bottom margin to this unlabeled list for better vertical flow. 

This is how it looks with `margin-bottom: 0.5em;` as an example:

Screen Shot 2020-03-31 at 6 08 02 PM

**Comment by rcanepa** [comment]
@ChaosExAnima , I might be missing something, but if I have JP Backup Real-Time and then I purchase JP Premium, the conflict notice doesn't appear. I only see the conflict notice when I have JP Backup Daily. Is that expected?

File: client/blocks/product-plan-overlap-notices/index.jsx (hunk 1 above)
Comment (ChaosExAnima) [reply client/blocks/product-plan-overlap-notices/index.jsx:135]: Yeah, noticed that too. Let me add that.

**Comment by ChaosExAnima** [comment]
@rcanepa - Will check that out! It should appear, yes.

File: client/blocks/product-plan-overlap-notices/index.jsx (hunk 1 above)
Comment (ChaosExAnima) [reply client/blocks/product-plan-overlap-notices/index.jsx:135]: Resolved with 506826a913fe5830e3c1122b02453add71e09466.

**Comment by ChaosExAnima** [comment]
Added bottom margin. @enejb, when you get in tomorrow can you give this a 👍?

**Comment by enejb** [comment]
@rcanepa  
> I might be missing something, but if I have JP Backup Real-Time and then I purchase JP Premium, the conflict notice doesn't appear. I only see the conflict notice when I have JP Backup Daily. Is that expected?

Yes that is correct Premium plan comes with the Daily Backup.

**Comment by enejb** [comment]
@ChaosExAnima can we add some more space between the `Consider removing conflicting products` and the list of products? 

See: 
Screen Shot 2020-04-02 at 5 20 56 PM

Also when I am on the single purchase view of the Search Product. I am still seeing the notice. 
My expectation is not to see it on that view. 
Screen Shot 2020-04-02 at 5 21 37 PM


Maybe that message can be more targeted as well and since the links lead to the same place we can remove them as well? 

cc: @keoshi

**Comment by ChaosExAnima** [comment]
@enejb - Have you done this that with the latest build? I added padding.

Re: single view, I can look.

**Comment by ChaosExAnima** [comment]
Okay, I've altered the behavior so that for single purchase views, it does a check for whether the product is part of the overlap. If not, the notice does not display. When it does display however, it will always show all overlapping products:

Screen Shot 2020-04-02 at 2 18 45 PM

Also added the styles so the padding is present:

Screen Shot 2020-04-02 at 2 18 33 PM

**Comment by keoshi** [comment]
This is looking really good. Thank you for making those styling tweaks, @ChaosExAnima

**Comment by ChaosExAnima** [comment]
🚢

**Pull request merged**